Replace sasl with pyhive.get_installed_sasl for pure-sasl compatibility

kesem0811 opened the pull request:
- Remove direct sasl/saslwrapper imports
- Use pyhive.hive.get_installed_sasl which handles pure-sasl compatibility
- Maintain same SASL authentication logic for GSSAPI/Kerberos
- Ensures compatibility with pure-sasl package while preserving functionality

<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [airflow-core/newsfragments](https://github.com/apache/airflow/tree/main/airflow-core/newsfragments).

kesem0811 committed:
Replace sasl with pyhive.get_installed_sasl for pure-sasl compatibility

- Remove direct sasl/saslwrapper imports
- Use pyhive.hive.get_installed_sasl which handles pure-sasl compatibility
- Maintain same SASL authentication logic for GSSAPI/Kerberos
- Ensures compatibility with pure-sasl package while preserving functionality

romsharon98 reviewed (approved):
LGTM, only static checks need to be fixed

kesem0811 committed:
fix ruff formating

eladkal commented on dags/simple_spark_xcom_test.py:
I assume this was for manual testing and should have not been committed?

kesem0811 replied:
You are right, I deleted those files.

kesem0811 committed:
cleanup: remove test DAG files to match UI version

- Remove dags/simple_spark_xcom_test.py
- Remove dags/test_spark_kubernetes_xcom.py
- Keep only the Hive provider changes for pure-sasl compatibility

kesem0811 committed:
Apply ruff formatting to hive.py

- Reorganize imports according to ruff standards
- Fix import ordering and formatting
- Ensure code follows ruff linting rules
- Auto-fix 2 ruff errors found in the file
